ingestion/src/metadata/ingestion/source/database/dynamodb.py

```diff
@@ -63,59 +68,76 @@ def create(cls, config_dict, metadata_config: OpenMetadataConnection):
     def prepare(self):
         pass
 
-    def next_record(self) -> Iterable[Entity]:
-        try:
-            table_list = list(self.dynamodb.tables.all())
-            if not table_list:
-                return
-            yield from self.ingest_tables()
-        except Exception as err:
-            logger.debug(traceback.format_exc())
-            logger.error(err)
+    def get_database_names(self) -> Iterable[str]:
+        """
+        Default case with a single database.
 
-    def ingest_tables(self, next_tables_token=None) -> Iterable[OMetaDatabaseAndTable]:
-        tables = list(self.dynamodb.tables.all())
-        for table in tables:
-            try:
+        It might come informed - or not - from the source.
+
+        Sources with multiple databases should overwrite this and
+        apply the necessary filters.
+        """
+
+        database_name = "default"
+        yield database_name
+
+    def yield_database(self, database_name: str) -> Iterable[CreateDatabaseRequest]:
+        """
+        From topology.
+        Prepare a database request and pass it to the sink
+        """
+
+        yield CreateDatabaseRequest(
+            name=database_name,
+            service=EntityReference(
+                id=self.context.database_service.id,
+                type="databaseService",
+            ),
+        )
+
+    def get_database_schema_names(self) -> Iterable[str]:
+        """
+        return schema names
+        """
+        if self.service_connection.__dict__.get("databaseSchema"):
+            yield self.service_connection.databaseSchema
+
+    def yield_database_schema(
+        self, schema_name: str
+    ) -> Iterable[CreateDatabaseSchemaRequest]:
+        """
+        From topology.
+        Prepare a database schema request and pass it to the sink
+        """
+
+        yield CreateDatabaseSchemaRequest(
+            name=schema_name,
+            database=EntityReference(id=self.context.database.id, type="database"),
+        )
+
+    def get_tables_name_and_type(self) -> Optional[Iterable[Tuple[str, str]]]:
+        """
+        Handle table and views.
+
+        Fetches them up using the context information and
+        the inspector set when preparing the db.
+
+        :return: tables or views, depending on config
+        """
+        schema_name = self.context.database_schema.name.__root__
+        if self.source_config.includeTables:
+            tables = list(self.dynamodb.tables.all())
```

Comment:
IDK if there's a way to pick up this information as a generator instead of casting a list. Might not be an issue, but would be more scalable

Reply:
Dynamodb gives us `dynamodb.tablesCollection(dynamodb.ServiceResource(), dynamodb.Table)` object which we can directly use instead of list casting